CMakeLists.txt

```diff
@@ -28,8 +28,14 @@ project(ats)
 
 enable_testing()
 set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+list(APPEND CMAKE_REQUIRED_DEFINITIONS -D_GNU_SOURCE)
```

Comment:
Is that going to be a problem for clang folk?

Reply:
This was to get some symbols from sys/types.h iirc.  I'm not sure why this wasn't pre-defined?  It might be better fixed another way eventually.